### PR TITLE
Redone journal.js model 

### PR DESCRIPTION
**Pull request by akashkhambay**

### CODE REVIEW

File: server/models/journal.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
-const fs = require('fs');
-const jsonData = fs.readFileSync('../data.json');
-const data = JSON.parse(jsonData);
+// const fs = require('fs');
+// const jsonData = fs.readFileSync('../data.json');
+// const data = JSON.parse(jsonData);
```
Comment (Velocima) [inline server/models/journal.js]: These can safely be removed.

File: server/models/journal.js [hunk 2]
```diff
@@ -9,33 +12,78 @@ class Journal {
 		this.id = journal.id;
 		this.comments = journal.comments || [];
```
Comment (Velocima) [inline server/models/journal.js:9]: emoji object can be added to constructor: ```

		this.emojis = journal.emojis || {
			likes: 0,
			loves: 0,
			dislikes: 0,
		};
```

File: server/models/journal.js [hunk 3]
```diff
@@ -9,33 +12,78 @@ class Journal {
...
+		const allData = readDataFromFile();
+		return allData;
+	}
+
 	static createNewJournalEntry(journal) {
 		const newJournalId = createNewId();
 		const newJournal = new Journal ({ id: newJournalId, ...journal});
 		data.push(newJournal);
 		return newJournal;
 	}
```
Comment (Velocima) [inline server/models/journal.js]: refactor to use new data methods

File: server/models/journal.js [hunk 4]
```diff
@@ -9,33 +12,78 @@ class Journal {
...
+	// 	);
+	// 	return allJournals;
+	// }
+
+	// static findJournalById(idToFind) {
+	// 	const {journal} = data.journals.filter(
+	// 		(journal) => idToFind === journal.id
+	// 	)[0];
+	// 	return new Journal();
+	// }
```
Comment (Velocima) [inline server/models/journal.js]: These can safely be removed.

File: server/models/journal.js [hunk 5]
```diff
@@ -9,33 +12,78 @@ class Journal {
...
+		const selectionOfEmojis = ['like', 'love', 'dislike'];
+		if (!selectionOfEmojis.includes(emoji)) throw new Error('Unavailable emojis');
+		const journalIndex = data.journals.find((journal) => journal.id === JournalId);
+		if (isIncrease) {
+			data.journals[journalIndex].emojis[emoji]++;
+		} else {
+			data.journals[journalIndex].emojis[emoji]--;
+		}
+		this.writeNewJournalDataToFile(data);
+		return true;
```
Comment (Velocima) [inline server/models/journal.js]: Return `new Journal(data.journals[journalIndex]);`

File: server/models/journal.js [hunk 6]
```diff
@@ -9,33 +12,78 @@ class Journal {
...
+	}
```
Comment (Velocima) [inline server/models/journal.js:80]: methods to add: 
- getAllCategories
- getAllJournalsInCategory
- updateJournal

File: server/models/journal.js (hunk 3 above)
Comment (akashkhambay) [reply server/models/journal.js]: Done :)

File: server/models/journal.js (hunk 5 above)
Comment (akashkhambay) [reply server/models/journal.js]: done :)

File: server/models/journal.js (hunk 4 above)
Comment (akashkhambay) [reply server/models/journal.js]: done :)

File: server/models/journal.js (hunk 2 above)
Comment (akashkhambay) [reply server/models/journal.js:9]: done :)

File: server/models/journal.js (hunk 6 above)
Comment (akashkhambay) [reply server/models/journal.js:80]: Bullet points 1 and 3 done :)

**Review by Velocima — COMMENTED** [review]
Nice one :) Merging to staging.

**Pull request merged**